zebra-chain/src/sapling/tests/prop.rs

```diff
@@ -99,12 +97,37 @@ proptest! {
         prop_assert_eq![tx, tx_parsed];
     }
 
+    /// Serialize and deserialize `SharedAnchor` shielded data by including it
+    /// in a V5 transaction
+    #[test]
+    fn shielded_data_v5_roundtrip(
+        shielded_v5 in any::<sapling::ShieldedData<SharedAnchor>>(),
+    ) {
+        zebra_test::init();
+
+        // shielded data doesn't serialize by itself, so we have to stick it in
+        // a transaction
+
+        // stick `SharedAnchor` shielded data into a v4 transaction
+        let tx = Transaction::V5 {
+            inputs: Vec::new(),
+            outputs: Vec::new(),
+            lock_time: LockTime::min_lock_time(),
+            expiry_height: block::Height(0),
+            sapling_shielded_data: Some(shielded_v5),
+            rest: Vec::new(),
+        };
```

Comment:
v5 shielded data should have a serialize implementation, because all its data is continuous.